### ISSUE TITLE
Puli install/build does not work in Sensiolabs Insight

### ISSUE THREAD

**Issue by pkruithof** [puli/issues#84]
Log from Sensiolabs Insight when it tried to install dependencies:
 
```
<warn>The "puli"/"puli.phar" command could not be found.</warn>
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
 
[...]
 
Generating optimized autoload files
 
Fatal error: Call to a member function run() on null in /home/foobar/code/vendor/puli/composer-plugin/src/PuliPlugin.php on line 428
```
 
Relevant composer.json lines:
 
```js
{
  "puli/cli":            "~1.0",
  "puli/symfony-bundle": "~1.0",
  "puli/twig-extension": "~1.0",
}
```

All dependencies are up to date, on my dev-machine everything works fine, and the lock file is committed so it's using the same versions. Unfortunately I don't know too much about the setup at Insight, it's a rather black boxed environment. But maybe this rings a bell.

**Comment by webmozart** [comment]
Thank you for reporting this issue @pkruithof! I can confirm this. Since "puli" is not installed on Insight, the command obviously fails. We'd need to install the puli.phar before. Any ideas?

**Comment by pkruithof** [comment]
But since Puli is in my composer file, would it not be possible to just use `./bin/puli`?

**Comment by webmozart** [comment]
Ah yes, that's possible of course :) I'm only talking about the case where Puli is installed as PHAR.

**Comment by webmozart** [comment]
Ok, I think I understand your problem. You installed Puli as Composer dependency, the "puli" binary is placed in bin/ (which is declared in composer.json as "bin-dir", right?), but still it's not found on Insight. Is this correct?

**Comment by pkruithof** [comment]
Yes that's correct.

For your first assumption the solution could be to download the phar in a `before_script` stage (I don't know the SI equivalent, but they have hooks like that). Maybe that would also work in this case but I assumed it should work with the dependency as well.

**Comment by webmozart** [comment]
Is the repository in question public? Can you reproduce the problem on some public repository?

**Comment by webmozart** [comment]
This should be fixed in 1.0.0-beta7.
